apis/compute/v1alpha1/user_types.go

```diff
@@ -0,0 +1,137 @@
+package v1alpha1
+
+import (
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// UserParameters defines the desired state of a User.
+// Required values when creating a User:
+// Administrator
+// Email
+// FirstName
+// ForceSecAuth
+// LastName
+// Password
+type UserParameters struct {
+	// Administrator The group has permission to edit privileges on this resource.
+	//
+	// +kubebuilder:validation:Required
+	Administrator bool `json:"administrator"`
+	// Email An e-mail address for the user.
+	//
+	// +kubebuilder:validation:Required
+	Email string `json:"email"`
+	// FirstName A first name for the user.
+	//
+	// +kubebuilder:validation:Required
+	FirstName string `json:"firstName"`
+	// ForceSecAuth Indicates if secure (two-factor) authentication should be enabled for the user (true) or not (false).
+	//
+	// +kubebuilder:validation:Required
+	ForceSecAuth bool `json:"forceSecAuth"`
+	// LastName A last name for the user.
+	//
+	// +kubebuilder:validation:Required
+	LastName string `json:"lastName"`
+	// Password A password for the user.
+	//
+	// +kubebuilder:validation:Required
+	Password string `json:"password"`
```

Comment:
Would be great if we could provide this from an env variable or secret. Something similar to what is being done for postgres_user, using CredentialsSource. Maybe in another CR.

Reply:
maybe we can add a todo comment to remember. thanks

Reply:
Super valid point. Thank you for raising this, I agree that is a much better way to address the password.